components/financial_assistance/app/models/financial_assistance/applicant.rb

```diff
@@ -291,7 +291,8 @@ class Applicant # rubocop:disable Metrics/ClassLength TODO: Remove this
     embeds_one :income_response, class_name: "EventResponse"
     embeds_one :mec_response, class_name: "EventResponse"
 
-    accepts_nested_attributes_for :incomes, :deductions, :benefits, :evidences
+    accepts_nested_attributes_for :incomes, :deductions, :benefits
+    accepts_nested_attributes_for :evidences, :reject_if => proc { |evidence| evidence[:key].blank? }, allow_destroy: true
```

Comment:
We suspect this might be a root cause so we're rejecting any nested evidences with blank key values